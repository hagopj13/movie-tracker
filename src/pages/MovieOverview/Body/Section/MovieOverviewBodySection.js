// @flow
import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

type Props = {
  title: string,
  children: Node,
};

const useStyles = makeStyles((theme) => ({
  root: {
    [theme.breakpoints.down('xs')]: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
    },
  },
  title: {
    marginBottom: theme.spacing(2),
  },
}));

const MovieOverviewBodySection = (props: Props) => {
  const { title, children } = props;

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography className={classes.title} variant="h5">
        {title}
      </Typography>
      {children}
    </div>
  );
};

export default MovieOverviewBodySection;
