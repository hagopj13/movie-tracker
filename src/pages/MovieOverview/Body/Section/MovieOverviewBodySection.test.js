import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Typography from '@material-ui/core/Typography';

import MovieOverviewBodySection from './MovieOverviewBodySection';

describe('MovieOverviewBodySection component', () => {
  let shallow;
  let title;
  let wrapper;

  const ChildComponent = () => <div>Some child component</div>;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    title = 'Some title';
    const mockProps = {
      title,
    };
    wrapper = shallow(
      <MovieOverviewBodySection {...mockProps}>
        <ChildComponent />
      </MovieOverviewBodySection>,
    );
  });

  it('should render the MovieOverviewBodySection component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the title', () => {
    expect(wrapper.find(Typography).text()).toBe(title);
  });

  it('should render the child component', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });
});
