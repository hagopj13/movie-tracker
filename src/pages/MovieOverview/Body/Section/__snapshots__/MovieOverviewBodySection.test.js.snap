// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieOverviewBodySection component should render the MovieOverviewBodySection component correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-title-2"
    variant="h5"
  >
    Some title
  </WithStyles(ForwardRef(Typography))>
  <ChildComponent />
</div>
`;
